subsys/net/ip/ipv6.c

```diff
@@ -1547,6 +1546,15 @@ static void nd_reachable_timeout(struct k_work *work)
 		return;
 	}
 
+	if (net_rpl_get_interface() && nbr->iface == net_rpl_get_interface()) {
```

Comment:
is it possible to have an nbr without an iface set?

If not, then you could remove the first part of this test.